e2e/specs/formsDashboard.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { test } from "../core";
+import { expect } from "@playwright/test";
+import { FormBuilderPage } from "../pages";
+import {
+  createForm,
+  createValueReference,
+  addFormResources,
+  deleteForm,
+} from "../commands/formOperations";
+import { Form } from "../../src/types";
+
+let form: Form = null;
+test.beforeEach(async ({ api }) => {
+  form = await createForm(api);
+  const valueReference = await createValueReference(api);
+  await addFormResources(api, valueReference, form.uuid);
+});
+
+test("should filter forms based on publish status", async ({ page }) => {
+  const formBuilderPage = new FormBuilderPage(page);
+  await formBuilderPage.gotoFormBuilder();
+
+  // Select the "unPublished" filter option
+  await formBuilderPage.publishStatusDropdownButton().click();
+
+  await formBuilderPage.unPublishedOption().click();
+
+  // Wait for the table to update with filtered results
+  await page.locator(".cds--data-table-container");
+
+  const tableRows = await page.$$eval(
+    '[data-testid^="form-row-"]',
+    (rows) => rows
+  );
+
+  expect(tableRows.length).toBeGreaterThan(1);
+
+  const row = await page.$(`[data-testid="form-row-0"]`);
+  const thirdTdElement = await row.$("td:nth-child(3)");
+
+  const publishStatusTag = await thirdTdElement.$(".cds--tag");
```

Comment:
@hadijahkyampeire Shall we use test id for this instead of the carbon class?

Reply:
The best practice would be using locators like `getByRole`, `getByLabelText` etc. But if it is not possible, we can use `getByTestId`